common4j/src/main/com/microsoft/identity/common/java/crypto/BasicDecryptor.java

```diff
@@ -66,20 +68,73 @@ public byte[] perform() throws ClientException {
         );
     }
 
+    @Override
+    public byte[] decrypt(@NonNull final Key key,
+                          @NonNull final String decryptAlgorithm,
+                          final byte[] iv,
+                          final byte[] dataToBeDecrypted,
+                          final int tagLength,
+                          final byte[] aad) throws ClientException {
+        return performCryptoOperationAndUploadTelemetry(
+                CryptoObjectName.Cipher,
+                decryptAlgorithm,
+                mCryptoFactory,
+                new ICryptoOperation<byte[]>() {
+                    @Override
+                    public byte[] perform() throws ClientException {
+                        return decryptWithCipher(
+                                key, decryptAlgorithm, iv, dataToBeDecrypted, tagLength, aad
+                        );
+                    }
+                }
+        );
+    }
+
     private byte[] decryptWithCipher(@NonNull final Key key,
                                      @NonNull final String decryptAlgorithm,
                                      final byte[] iv,
                                      byte[] dataToBeDecrypted)
             throws ClientException {
         final Cipher cipher = mCryptoFactory.getCipher(decryptAlgorithm);
-        try{
+        try {
             if (iv != null && iv.length > 0) {
                 final IvParameterSpec ivSpec = new IvParameterSpec(iv);
                 cipher.init(Cipher.DECRYPT_MODE, key, ivSpec);
             } else {
                 cipher.init(Cipher.DECRYPT_MODE, key);
             }
-                return cipher.doFinal(dataToBeDecrypted);
+            return cipher.doFinal(dataToBeDecrypted);
+        } catch (final BadPaddingException e) {
+            throw new ClientException(ClientException.BAD_PADDING, e.getMessage(), e);
+        } catch (final IllegalBlockSizeException e) {
+            throw new ClientException(ClientException.INVALID_BLOCK_SIZE, e.getMessage(), e);
+        } catch (final InvalidKeyException e) {
+            throw new ClientException(ClientException.INVALID_KEY, e.getMessage(), e);
+        } catch (final InvalidAlgorithmParameterException e) {
+            throw new ClientException(ClientException.INVALID_ALG_PARAMETER, e.getMessage(), e);
+        }
+    }
+
+    @SuppressWarnings("NewApi")
+    private byte[] decryptWithCipher(@NonNull final Key key,
```

Comment:
decryptWithGcm

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1953/commits/8ef8a06a8f8c110c7e06918e087f97256b0ef69c